src/espruino/projects/Sensors/DistanceSensor-Advanced-WidoraAir.js

```diff
@@ -0,0 +1,234 @@
+    if (newValue !== onBoardLedValue) {
+      onBoardLedValue = newValue;
+      digitalWrite(onBoardLedPin, onBoardLedValue);
+    }
+  };
+
+  const onWifiConnecting = () => {
+    print('[Wifi] [Info] Connecting...');
+    switchOnBoardLed(onBoardLedOnValue);
+  };
+
+  const onWifiConnected = () => {
+    print('[Wifi] [Info] Connected:', wifi.getIP());
+    mqttClient.connect();
+  };
+
+  const connect = () => {
+    const connection = wifi.getDetails();
+    if (connection.status === 'connected' && connection.ssid === settings.ssid) {
+      onWifiConnected();
+      return;
+    }
+
+    onWifiConnecting();
+    wifi.connect(settings.ssid, { password: settings.wifiPassword });
+  };
+
+  wifi.on('connected', () => {
+    onWifiConnected();
+  });
+
+  wifi.on('disconnected', (d) => {
+    print('[Wifi] [Error] No connection, details:', d);
+    mqttClient.disconnect();
+    connect();
+  });
+
+  const sendTelemetry = (sensorId, inRange) => {
+    let telemetry = {
+      freeMemory: process.memory().free
+    };
+
+    mqttClient.publish(getTopic('telemetry/thing'), JSON.stringify(telemetry), 1);
+
+    if (sensorId !== undefined) {
+      telemetry = {
+        "sensorId": sensorId,
+        "inRange": inRange
+      };
+
+      mqttClient.publish(getTopic('telemetry/sensor'), JSON.stringify(telemetry), 1);
+    }
+  };
+
+  mqttClient.on('connected', () => {
+    switchOnBoardLed(!onBoardLedOnValue);
+    mqttClient.subscribe(getTopic('command'), 1);
+
+    mqttClient.publish(
+      getTopic('status'),
+      JSON.stringify(getStatusMessage(true)),
+      1,
+      true
+    );
+
+    telemetryInterval = setInterval(() => sendTelemetry(), 30 * 1000);
+    sendTelemetry();
+  });
+
+  mqttClient.on('disconnected', () => {
+    switchOnBoardLed(onBoardLedOnValue);
+
+    if (telemetryInterval !== -1) {
+      clearInterval(telemetryInterval);
+      telemetryInterval = -1;
+    }
+  });
+
+  const doSomething = () => {
+  };
+
+  mqttClient.on('receive', (message) => {
+    print('[Mqtt] [Info] Incoming message:', message);
+    const command = JSON.parse(message.content);
+
+    if (command.name === 'doSomething') doSomething();
+  });
+
+  // mqttClient.on("debug", debug => {
+  //   print("[Mqtt] [Debug]", debug);
+  // });
+
+  mqttClient.on('info', (info) => {
+    print('[Mqtt] [Info]', info);
+  });
+
+  mqttClient.on('error', (error) => {
+    print('[Mqtt] [Error]', error);
+  });
+
+  connect();
+
+  global.settings = settings;
+  global.wifi = wifi;
+  global.mqttClient = mqttClient; 
+
+  const numberOfMeasurements = 10;
+  const measurements1 = new Array(numberOfMeasurements);
+  let nextMeasurement = 0;
+  const minNumberInRange = 5;
+  const minNumberOutOfRange = 4;
+  let lastQueueState = false;
+  let queueDetected = false;
+  
+  print('Connecting to HC-SR04 sensor 1...');
+  const sensor1 = require('HC-SR04').connect(D21, D22, (dist) => {
+      const inRange = (dist <= 200);
```

Comment:
Krijg je nooit een hele hoge waarde als mensen in range zijn? Dat zag ik gebeuren toen ik met de sensor experimenteerde.

Reply:
We zijn niet lang genoeg met de data aan de slag geweest om daar een goed beeld van te krijgen, maar de waarden waren soms wel all over the place.